From a08b7f83778f02abcc95de1bd2e767cde3bb314c Mon Sep 17 00:00:00 2001
From: Zumeng Chen <zumeng.chen@windriver.com>
Date: Mon, 26 Oct 2015 09:48:00 +0800
Subject: [PATCH 292/292] zynq-edac: probe_quit when there is no EDAC device
 accordingly.

This patch is to get rid of the error information when there is actually
no EDAC device in some specific ZYNQ boards.

Signed-off-by: Hu <yadi.hu@windriver.com>
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/edac/zynq_edac.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/drivers/edac/zynq_edac.c b/drivers/edac/zynq_edac.c
index 9225ca2..fafdac8 100644
--- a/drivers/edac/zynq_edac.c
+++ b/drivers/edac/zynq_edac.c
@@ -16,6 +16,7 @@
  */
 
 #include <linux/edac.h>
+#include <linux/of.h>
 #include <linux/module.h>
 #include <linux/platform_device.h>
 
@@ -504,6 +505,15 @@ static int zynq_edac_mc_probe(struct platform_device *pdev)
 	int rc;
 	struct resource *res;
 	void __iomem *baseaddr;
+	struct device_node *np = pdev->dev.of_node;
+	u32 *edac_enable;
+
+	if (!np || !of_device_is_available(np))
+		return -ENODEV;
+
+	edac_enable = (u32 *)of_get_property(np, "xlnx,has-ecc", NULL);
+        if (!edac_enable || !(*edac_enable))/* property has-ecc disenabled or no exist */
+		return -ENODEV;
 
 	/* Get the data from the platform device */
 	res = platform_get_resource(pdev, IORESOURCE_MEM, 0);
-- 
2.0.2

